src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php

```diff
@@ -193,150 +253,180 @@ public function daemon(InputInterface $input, OutputInterface $output)
      */
-    public function requestShutdown()
+    protected function dispatchEvent($eventName)
     {
-        $this->shutdownRequested = true;
+        $dispatcher = $this->getEventDispatcher();
+        if (!is_null($dispatcher)) {
+            $time = !is_null($this->startTime) ? microtime(true) - $this->startTime : 0;
+
+            $event = new DaemonEvent($this);
+            $event->setExecutionTime($time);
+
+            $dispatcher->dispatch($eventName, $event);
+        }
 
         return $this;
     }
 
     /**
-     * Is shutdown requested.
+     * get the EventDispatcher.
      *
-     * @return bool
+     * @return object|EventDispatcherInterface
      */
-    public function isShutdownRequested()
+    public function getEventDispatcher()
     {
-        return $this->shutdownRequested;
+        return $this->dispatcher;
     }
 
-    /**
-     * @param int $v setter for LoopCount
-     *
-     * @return $this
-     */
-    public function setLoopCount($v)
+    protected function setup(InputInterface $input, OutputInterface $output)
     {
-        $this->loopCount = (int) $v;
-
-        return $this;
     }
 
     /**
-     * @return int
+     * @param InputInterface $input
+     * @param OutputInterface $output
      */
-    public function getLoopCount()
+    protected function loop(InputInterface $input, OutputInterface $output)
     {
-        return $this->loopCount;
+        $this->startTime = microtime(true);
+
+        try {
+            // Loop Callback
+            call_user_func($this->loopCallback, $input, $output);
+
+            // Loop interval callback
+            $this->callIterationsIntervalCallbacks($input, $output);
+        } catch (StopLoopException $e) {
+            $this->setLastException($e);
+            $this->dispatchEvent(DaemonEvents::DAEMON_LOOP_EXCEPTION_STOP);
+
+            $this->returnCode = $e->getCode();
+            $this->requestShutdown();
+        } catch (\Exception $e) {
+            $this->setLastException($e);
+            $this->dispatchEvent(DaemonEvents::DAEMON_LOOP_EXCEPTION_GENERAL);
+
+            if ($this->getShowExceptions()) {
+                $this->getApplication()->renderException($e, $output);
+            }
+
+            if ($this->getShutdownOnException()) {
+                $this->returnCode = !is_null($e->getCode()) ? $e->getCode() : -1;
+                $this->requestShutdown();
+            }
+        }
+
+        $this->incrLoopCount();
+        $this->dispatchEvent(DaemonEvents::DAEMON_LOOP_ITERATION);
+        $this->dispatchConfigurationEvents();
+
+        if (!$this->isLastLoop()) {
+            if ($this->sleepingTime) {
+                // We will wake up later but the event loop is still active for background operations
+                $this->loop->addTimer($this->sleepingTime, function () use ($input, $output) {
+                    $this->loop($input, $output);
+                });
+
+                $this->sleepingTime = 0;
```

Comment:
if we reset `sleepingTime` here, it means that the user of this bundle will have to set `sleepingTime` after each loop?

(maybe I should have reviewed this after the documentation was available 🤦‍♂️ )

Reply:
That's the idea - please see my previous comment (and so we must change the name of  setSleepingTime)